sdk/python/feast/client.py

```diff
@@ -353,6 +360,145 @@ def archive_project(self, project):
         if self._project == project:
             self._project = FEAST_DEFAULT_OPTIONS[CONFIG_PROJECT_KEY]
 
+    def apply_entity(
+        self, entities: Union[List[EntityV2], EntityV2], project: str = None
+    ):
+        """
+        Idempotently registers entities with Feast Core. Either a single
+        entity or a list can be provided.
+
+        Args:
+            entities: List of entities that will be registered
+
+        Examples:
+            >>> from feast import Client
+            >>> from feast.entity import EntityV2
+            >>> from feast.value_type import ValueType
+            >>>
+            >>> feast_client = Client(core_url="localhost:6565")
+            >>> entity = EntityV2(
+            >>>     name="driver_entity",
+            >>>     description="Driver entity for car rides",
+            >>>     value_type=ValueType.STRING,
+            >>>     labels={
+            >>>         "key": "val"
+            >>>     }
+            >>> )
+            >>> feast_client.apply_entity(entity)
+        """
+
+        if project is None:
+            if self.project is not None:
+                project = self.project
+            else:
+                raise ValueError("No project has been configured.")
+
+        if not isinstance(entities, list):
+            entities = [entities]
+        for entity in entities:
+            if isinstance(entity, EntityV2):
+                self._apply_entity(project, entity)
+                continue
+            raise ValueError(f"Could not determine entity type to apply {entity}")
+
+    def _apply_entity(self, project: str, entity: EntityV2):
+        """
+        Registers a single entity with Feast
+
+        Args:
+            entity: Entity that will be registered
+        """
+
+        entity.is_valid()
+        entity_proto = entity.to_proto()
+
+        # Convert the entity to a request and send to Feast Core
+        try:
+            apply_entity_response = self._core_service.ApplyEntity(
+                ApplyEntityRequest(project=project, entity=entity_proto),
+                timeout=self._config.getint(CONFIG_GRPC_CONNECTION_TIMEOUT_DEFAULT_KEY),
+                metadata=self._get_grpc_metadata(),
+            )  # type: ApplyEntityResponse
+        except grpc.RpcError as e:
+            raise grpc.RpcError(e.details())
+
+        # Extract the returned entity
+        applied_entity = EntityV2.from_proto(apply_entity_response.entity)
+
+        # Deep copy from the returned entity to the local entity
+        entity._update_from_entity(applied_entity)
+
+    def list_entities_v2(
```

Comment:
I guess in SDK we can replace previous methods, as we discussed

Reply:
Replaced.